In [18]:
import pandas as pd
import seaborn as sns

In [19]:
life = pd.read_csv('lifeExpectancyAtBirth.csv', sep=',')

In [20]:
life.head()

Location  Period                         Indicator        Dim1  \
0  Afghanistan    2019  Life expectancy at birth (years)  Both sexes   
1  Afghanistan    2019  Life expectancy at birth (years)        Male   
2  Afghanistan    2019  Life expectancy at birth (years)      Female   
3  Afghanistan    2015  Life expectancy at birth (years)  Both sexes   
4  Afghanistan    2015  Life expectancy at birth (years)        Male   

   First Tooltip  
0          63.21  
1          63.29  
2          63.16  
3          61.65  
4          61.04

In [21]:
filterLife = life['Period'] == 2019

In [22]:
life[filterLife].head()

Location  Period                         Indicator        Dim1  \
0   Afghanistan    2019  Life expectancy at birth (years)  Both sexes   
1   Afghanistan    2019  Life expectancy at birth (years)        Male   
2   Afghanistan    2019  Life expectancy at birth (years)      Female   
12      Albania    2019  Life expectancy at birth (years)  Both sexes   
13      Albania    2019  Life expectancy at birth (years)        Male   

    First Tooltip  
0           63.21  
1           63.29  
2           63.16  
12          78.00  
13          76.25

In [23]:
filterSex = life['Dim1'] == 'Both sexes'

In [24]:
life[filterSex]

Location  Period                         Indicator        Dim1  \
0     Afghanistan    2019  Life expectancy at birth (years)  Both sexes   
3     Afghanistan    2015  Life expectancy at birth (years)  Both sexes   
6     Afghanistan    2010  Life expectancy at birth (years)  Both sexes   
9     Afghanistan    2000  Life expectancy at birth (years)  Both sexes   
12        Albania    2019  Life expectancy at birth (years)  Both sexes   
...           ...     ...                               ...         ...   
2182       Zambia    2000  Life expectancy at birth (years)  Both sexes   
2185     Zimbabwe    2019  Life expectancy at birth (years)  Both sexes   
2188     Zimbabwe    2015  Life expectancy at birth (years)  Both sexes   
2191     Zimbabwe    2010  Life expectancy at birth (years)  Both sexes   
2194     Zimbabwe    2000  Life expectancy at birth (years)  Both sexes   

      First Tooltip  
0             63.21  
3             61.65  
6             59.94  
9             54.99  
12            78.00  
...             ...  
2182          44.46  
2185          60.68  
2188          58.48  
2191          51.49  
2194          46.57  

[733 rows x 5 columns]

In [25]:
life[filterLife & filterSex].head()

Location  Period                         Indicator        Dim1  \
0           Afghanistan    2019  Life expectancy at birth (years)  Both sexes   
12              Albania    2019  Life expectancy at birth (years)  Both sexes   
24              Algeria    2019  Life expectancy at birth (years)  Both sexes   
36               Angola    2019  Life expectancy at birth (years)  Both sexes   
48  Antigua and Barbuda    2019  Life expectancy at birth (years)  Both sexes   

    First Tooltip  
0           63.21  
12          78.00  
24          77.13  
36          63.06  
48          76.45

In [26]:
filtered = life[filterLife & filterSex]

In [27]:
filtered.head()

Location  Period                         Indicator        Dim1  \
0           Afghanistan    2019  Life expectancy at birth (years)  Both sexes   
12              Albania    2019  Life expectancy at birth (years)  Both sexes   
24              Algeria    2019  Life expectancy at birth (years)  Both sexes   
36               Angola    2019  Life expectancy at birth (years)  Both sexes   
48  Antigua and Barbuda    2019  Life expectancy at birth (years)  Both sexes   

    First Tooltip  
0           63.21  
12          78.00  
24          77.13  
36          63.06  
48          76.45

In [28]:
filtered.min()

Location                              Afghanistan
Period                                       2019
Indicator        Life expectancy at birth (years)
Dim1                                   Both sexes
First Tooltip                               50.75
dtype: object

Laagste levensverwachting is dus in Afghanistan voor 2019 en beide geslachten

In [29]:
filtered['First Tooltip'].max()

84.26

Hoogste levensverwachting is dus 84.26 voor 2019 en beide geslachten

In [30]:
filtered['First Tooltip'].mean()

72.54049180327873

We weten dus dat de gemiddelde levensverwachting ongeveer 72,5 jaar oud is voor beide geslachten in 2019

In [31]:
filtered['First Tooltip'].median()

73.74

In [32]:
filtered.head()

Location  Period                         Indicator        Dim1  \
0           Afghanistan    2019  Life expectancy at birth (years)  Both sexes   
12              Albania    2019  Life expectancy at birth (years)  Both sexes   
24              Algeria    2019  Life expectancy at birth (years)  Both sexes   
36               Angola    2019  Life expectancy at birth (years)  Both sexes   
48  Antigua and Barbuda    2019  Life expectancy at birth (years)  Both sexes   

    First Tooltip  
0           63.21  
12          78.00  
24          77.13  
36          63.06  
48          76.45

In [33]:
filtered.sort_values('First Tooltip', ascending=False).head()

Location  Period                         Indicator        Dim1  \
997               Japan    2019  Life expectancy at birth (years)  Both sexes   
1897        Switzerland    2019  Life expectancy at birth (years)  Both sexes   
1573  Republic of Korea    2019  Life expectancy at birth (years)  Both sexes   
1837              Spain    2019  Life expectancy at birth (years)  Both sexes   
1753          Singapore    2019  Life expectancy at birth (years)  Both sexes   

      First Tooltip  
997           84.26  
1897          83.45  
1573          83.30  
1837          83.22  
1753          83.22

Dit betekent dat voor 2019 en beide geslachten deze 5 landen de hoogste levensverwachting produceren

In [34]:
filtered.sort_values('First Tooltip').head()

Location  Period                         Indicator  \
1117                   Lesotho    2019  Life expectancy at birth (years)   
373   Central African Republic    2019  Life expectancy at birth (years)   
1801                   Somalia    2019  Life expectancy at birth (years)   
661                   Eswatini    2019  Life expectancy at birth (years)   
1333                Mozambique    2019  Life expectancy at birth (years)   

            Dim1  First Tooltip  
1117  Both sexes          50.75  
373   Both sexes          53.10  
1801  Both sexes          56.47  
661   Both sexes          57.73  
1333  Both sexes          58.14

Dit betekent dat voor 2019 en beide geslachten deze 5 landen de laagste levensverwachting produceren